File: hw/codena/u3/hw23AbstractFactoryCode/Python/widgets/model/gui_factory.py
from abc import ABC, abstractmethod

from controller.config_reader import ConfigReader

class GUIFactory(ABC):

    @staticmethod
    def getFactory():

        from model.win_factory import WinFactory
        from model.linux_factory import LinuxFactory

        sys = ConfigReader.readFromConfigFile("OS_TYPE")

        if sys == 0:
            return WinFactory()
        else:
            return LinuxFactory()

    @abstractmethod
    def createButton(self):
        pass

    @abstractmethod
    def createMenu(self):
        pass

File: hw/codena/u3/hw23AbstractFactoryCode/Javascript/widgets/model/GUIFactory.js
import ConfigReader from "../controller/ConfigReader.js";

export default class GUIFactory {

    static async getFactory() {

        const { default: WinFactory } = await import("./WinFactory.js");
        const { default: LinuxFactory } = await import("./LinuxFactory.js");

        const sys = ConfigReader.readFromConfigFile("OS_TYPE");

        if (sys === 0) {
            return new WinFactory();
        }

        return new LinuxFactory();
    }

    createButton() {
        throw new Error("Must override");
    }

    createMenu() {
        throw new Error("Must override");
    }
}